### ISSUE TITLE
torch.no_grad on net?

### ISSUE THREAD

**Issue by TKassis** [lucidrains/ema-pytorch#25]
Thanks for an extremely useful package! Quick question regarding the example usage in the Readme. When you are mutating the network why are you disabling grad, don’t you want the ‘net’ network to learn then copy the weights to the EMA network? Am I misunderstanding this?

with torch.no_grad():
    net.weight.copy_(torch.randn_like(net.weight))
    net.bias.copy …..

**Comment by lucidrains** [comment]
@TKassis hey Timothy! good to hear from you!

yes indeed, it is just an example to show that it picks up on any changes

you can substitute that with your regular gradient descent step

**Comment by TKassis** [comment]
Got it!
